api/src/main/java/org/openmrs/api/impl/AdministrationServiceImpl.java

```diff
@@ -857,5 +858,23 @@ public boolean isDatabaseStringComparisonCaseSensitive() {
 	public void updatePostgresSequence() {
 		dao.updatePostgresSequence();
 	}
-	
+
+	@Override
+	public List<String> getSerializerWhitelistTypes() {
+		List<GlobalProperty> gpTypes = getGlobalPropertiesBySuffix(
+			AdministrationService.GP_SUFFIX_SERIALIZER_WHITELIST_TYPES);
+		List<String> whitelistTypes = new ArrayList<>();
+		whitelistTypes.add("org.openmrs.**");
```

Comment:
So, one of the things we'd ideally want to prevent is serializing or deserializing `org.openmrs.util.OpenmrsClassLoader`.

What would you think of a rule like:

```
xstream.allowTypeHierachy(OpenmrsObject.class);
// may be others, but these are the common exceptions to the above:
xstream.allowTypes(LayoutTemplate.class, ComplexData.class);
```

Reply:
Yeah, I was thinking of any classes that may be risky in org.openmrs, but I didn't come up with an obvious OpenmrsClassLoader... It seems that it may be better indeed to be less permissive and do as you suggested. I'll need to check what is needed for the Reporting module so that most obvious use cases are covered out of the box.